Duplicati/Library/Utility/ReadLimitLengthStream.cs

```diff
@@ -35,12 +66,12 @@ public override void Flush()
             // NOOP
         }
 
-        public override long Length => m_endPosition;
+        public override long Length => m_length;
 
         public override long Position
         {
-            get => m_innerStream.Position;
-            set => m_innerStream.Position = value;
+            get => ClampInnerStreamPosition(m_innerStream.Position) - m_start;
+            set => m_innerStream.Position = ClampInnerStreamPosition(value + m_start);
```

Comment:
I think most implementations (e.g., `FileStream`, `MemoryStream`, etc.) disallow `value` to be negative.  This seems to allow for it, and we also have a test for the case `value = -1`.  Is there a reason for this (possibly related to my [other comment](https://github.com/duplicati/duplicati/pull/4469/files#r597851972))?

Reply:
See the other comment - I've changed this and made the behavior when the base stream is at a Position before the starting point consistent with that.

Reply:
(Thanks for all the review comments, by the way!)